### PR TITLE
[6x Backport] Fix gprecoverseg with unreachable failed host

### PR DESCRIPTION
**Pull request by ShirishaRao**
This is a backport of [#13323](https://github.com/greenplum-db/gpdb/pull/13323)

 **Issue:**
 gprecoverseg -i fails to recover any segment if the failed host is unreachable.
 This is irrespective of full/incremental recovery, if the failed host is down then none of the down segments are recovered.
 Ideally the recovery should depend on the failover host status.

 **RCA:**
 Currently while recovering the segments with input configuration file, we check the status of failed host for both the below scenarios,
 1. when failed and failover host are same (in-place recovery)
 2. When failed and failover host are different
If the failed host is unreachable, recovery fails.

**Fix:**
Checking the status of failed host is done only in case of in-place recovery, specifically, when the failover information is not provided in the recover_config_file

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: gpMgmt/bin/gppylib/programs/clsRecoverSegment_triples.py [hunk 1]
```diff
@@ -269,12 +278,12 @@ def _convert_requests_to_triplets(self, requests):
                 failover.setSegmentAddress(req.failover_host_address)
                 failover.setSegmentPort(int(req.failover_port))
                 failover.setSegmentDataDirectory(req.failover_datadir)
-                failover.unreachable = False if req.failover_to_new_host else failover.unreachable
-
-            # this must come AFTER the if check above because failedSegment can be adjusted to
-            #   point to a different object
-            if req.failed.unreachable and not req.failover_to_new_host:
-                continue
+                failover.unreachable = failover.getSegmentHostName() in unreachable_failover_hosts
```
Comment (Annu149) [inline gpMgmt/bin/gppylib/programs/clsRecoverSegment_triples.py:281]: If failover is unreachable then also we should skip the recovery right? Can't see that part here. Will that be covered in another PR ?

**Review by Annu149 — APPROVED** [review]
Looks good to me

File: gpMgmt/bin/gppylib/programs/clsRecoverSegment_triples.py (hunk 1 above)
Comment (ShirishaRao) [reply gpMgmt/bin/gppylib/programs/clsRecoverSegment_triples.py:281]: Code for skipping the recovery when failover is unreachable is existing as part of validate method of 'RecoveryTriplet' class

**Review by piyushc01 — APPROVED** [review]
Looks good to me.

**Review by nimish350 — APPROVED** [review]
